fairing/preprocessors/converted_notebook.py

```diff
@@ -65,16 +65,20 @@ def __init__(self,
 
         self.notebook_file = notebook_file
         self.notebook_preprocessor = notebook_preprocessor
+        self._not_overwrite = False
 
     def preprocess(self):
+        if self._not_overwrite:
```

Comment:
If overwrite is False and the file exists lets throw an exception rather than just assuming the existing file is up to date.

Reply:
`self._not_overwrite` is initialized to be `False`.
After the first pass of `preprocess()`, `self._not_overwrite` is set to `True` no matter what the output path is (e.g., notebook_file.py or a temporary file, which is stored as `self.executable`).
And if we call `preprocess()` for multiple times, we do not output the file again because of this check `if self._not_overwrite:`.

It's a way to convert notebook to python script even though the default path has existed and the desired behavior you mentioned below is a way to confirm the output path does not change.
How do you think which one is better?